### PR TITLE
[Gtk Pipeline] General Improvements #4457

### PR DESCRIPTION
**Pull request by cra0zy**
This PR fixes stuff mentioned in #4457 for Gtk Pipeline Tool.

Stuff done:
 - Added "Open With", Linux only
 - Added Undo/Redo toolbar buttons
 - Added Cancel toolbar button which appears while the build is running, and build/rebuild buttons hide
 - Updated reference dialog to mimic WinForms one
 - Updated how reference property is displayed in propertygrid
 - Made headerbar separator invisible in case build/rebuild/cancel build aren't visible
 - Allowed user scrolling of build output possible
 - Smoother scrolling
 - Fixed opening of files that contain spaces

Update no. 1:
 - Toolbar items / Headerbar items are now connected to actions, this means that changing something on action will change them as well
 - Headerbar now has a proper menu

Update no. 2:
 - Fixed pressing open and then canceling creating an empty base item
 - Separate "Exclude from Project" and "Delete" options
 - Made build output cursor invisible (not mouse cursor, text cursor, text can still be selected)

### CODE REVIEW

File: Tools/Pipeline/Gtk/MainWindow.cs [hunk 1]
```diff
@@ -742,11 +742,26 @@ public void OnRenameActionActivated (object sender, EventArgs e)
...
         {
-            projectview1.Remove ();
-            UpdateMenus();
+            var dialog = new MessageDialog(this, DialogFlags.Modal, MessageType.Warning, ButtonsType.None, "How do you want to remove selected file/s?");
+            dialog.Title = "Delete File/s";
+            dialog.SecondaryText = "Do note that selecting \"Delete\" option will delete the file and this action will not be undoable.";
+
+            dialog.AddButton("Delete", ResponseType.Apply);
+            dialog.AddButton("Cancel", ResponseType.Cancel);
+            dialog.AddButton("Remove", ResponseType.Ok);
```
Comment (tomspilman) [inline Tools/Pipeline/Gtk/MainWindow.cs]: The alternative implementation is having a separate Delete and Remove commands sort of like VS.

I do worry that mixing them both together like this... then having a 3 button prompt... could lead to costly mistakes by users.

A remove which works without prompts and a delete that asks a "Are you sure this is not undoable"... seems like a safer approach.

@dellis1972 @KonajuGames ?

Comment (cra0zy) [inline Tools/Pipeline/Gtk/MainWindow.cs]: MonoDevelop/Xamarin Studio has this type of delete dialog:

deletedilalog

**Comment by cra0zy** [comment]
I am gonna use this PR to test something.

**Comment by cra0zy** [comment]
I am done with testing. One of the things I tested was Gtk Pipeline on Windows, with Gtk 3 the window looks very good, but the menus in menubar are awful, and with Gtk 2 menubar and toolbar background colors are off, which isn't that hard to override.

Gtk on Windows it's 32 bit only, so I wouldn't use it, this was more of a test to see how it looks.

**Comment by cra0zy** [comment]
@tomspilman I've done what you've asked.

**Comment by cra0zy** [comment]
@dellis1972 @tomspilman Can this be merged so I can start working on Win Forms side of things for #4457 ?

**Comment by cra0zy** [comment]
Bummer :|

**Comment by tomspilman** [comment]
Sorry... just been busy with work stuff.  I'll get to it tonight for sure.

**Comment by cra0zy** [comment]
I noticed that CollectionEditorDIalog had tabs left, so I just replaced them with spaces.

**Comment by cra0zy** [comment]
> Bummer :|

Agreed :P

**Comment by cra0zy** [comment]
I rebases this so it builds with the new build system.

**Comment by cra0zy** [comment]
@tomspilman @KonajuGames @dellis1972 Can this be merged for 3.5.1? I am tired of recompiling it each time I want to update system MG...

**Comment by KonajuGames** [comment]
Added it to the 3.5.1 milestone label so it doesn't get missed.

**Comment by cra0zy** [comment]
Thanks

**Comment by tomspilman** [comment]
I've moved this to 3.6...  we will wrap up this 3.5 service release this week, but that shouldn't keep this from being merged.

**Comment by cra0zy** [comment]
Since 3.5.1 got merged, can this get merged now? I really really really want to start working on what's discussed in #4238

**Comment by tomspilman** [comment]
Thanks @cra0zy !

**Pull request merged**